Add gravatar system with name initials fallback

alireza-ahmadi opened the pull request:
Thanks to @bengott's amazing library, I have implemented avatar(Gravatar with two character initials fallback) system for Libreboard, Hope it works.
![avatar](https://cloud.githubusercontent.com/assets/1054228/5726260/39026096-9b6f-11e4-8d15-72579f968371.png)

mquandalle commented on client/lib/utils.js:
```diff
@@ -142,6 +142,14 @@ Utils = {
+    },
+
+    // Extract userId from user object
+    getUserId: function(user) {
```
I think we don't want this function. In the code we should always know if a variable `x` is a user object (in this case we just use `x._id`) or an id (`x`). This function can bring confusion during the development.

alireza-ahmadi replied:
I have removed this method. Now developers can obtain user ID using `this.user._id`. Also in last commit I have replaced `this.user._id` with `this.userId` to make it more similar to original code, If you are OK with `this.user._id`, you can ignore last commit. (Sorry for my weak English grammar ;))

alireza-ahmadi replied:
I have removed this method. Now developers can obtain user ID using `this.user._id`. Also in last commit I have replaced `this.user._id` with `this.userId` to make it more similar to original code, If you are OK with `this.user._id`, you can ignore last commit. (Sorry for my weak English grammar ;))

mquandalle commented on client/views/users/helpers.js:
```diff
@@ -1,9 +1,16 @@
+
+Avatar.options = {
+    fallbackType: 'initials'
+};
```
Please add a new line at the end of the file.

mquandalle commented on client/views/users/templates.html:
```diff
@@ -185,12 +178,15 @@ <h3 class="bottom" style="margin-right: 40px;">
+        {{/if }}
     </div>
-</template>
+</template>
```
new line

mquandalle commented on client/views/users/templates.html:
```diff
@@ -185,12 +178,15 @@ <h3 class="bottom" style="margin-right: 40px;">
+        {{#if showStatus }}
+            <span class="member-status {{# if userData.profile.status}}active{{/if}}"></span>
+        {{/if }}
+        {{#if showMemberType}}
```
It seems that every time you call `{{> userAvatar }}` with the `showStatus` option, you also use the `showMemberType`. Maybe we could have only one option that would display the two icons. (:+1: for the option BTW)

alireza-ahmadi replied:
Two options replaced with single `showBadges` options.

mquandalle commented on client/views/widgets/events.js:
```diff
@@ -45,7 +45,7 @@ var getMemberIndex = function(board, searchId) {
     'click .js-change-role': function(event, t) {
         var currentBoard = Boards.findOne();
-        var memberIndex = getMemberIndex(currentBoard, this.memberId);
+        var memberIndex = getMemberIndex(currentBoard, Utils.getUserId(this.user));
```
I believe this change is not desirable.

mquandalle commented:
Hello @alireza-ahmadi,

Thank you for your great work. Please consider my inline comments before I can merge this PR.

mquandalle commented on client/views/users/templates.html:
```diff
@@ -81,10 +79,7 @@ <h1 id="landingLogo">
-                    {{#if isTrue currentUser.username profile.username }}
-                        <span class="change-text">{{_ 'change-avatar'}}</span>
-                    {{ /if }}
+                    {{> userAvatar user=profile size="large" removeWrapper=true}}
```
I think the name `noWrap` is better than `removeWrapper`.

alireza-ahmadi commented:
I have added the commit email to my Github account, Also I have fixed items that you mentioned in the comments. Please let me know if there is any other problems with this pull request.

mquandalle commented:
Thank you very much. I merged your changes in df7e69b3c41d.

mquandalle commented on client/views/users/member.styl:
```diff
@@ -16,15 +16,72 @@
     user-select: none;
     z-index: 0;
     text-decoration: none;
+    
```
BTW there was some trailing whitespaces here; we try to avoid them :-)

alireza-ahmadi replied:
Thanks, I will consider this point in future contributions.
